File: pi_code/pi_camera/camera_test.py

import cv2 as cv
import numpy as np
import math
import os

# 读取摄像机
path = "rtsp://192.168.137.135/11"
cap = cv.VideoCapture(path)
i = 0
while True:
    ret, frame = cap.read()
    cv.imshow("frame", frame)
    i = i + 1
    print("帧数", i)
    c = cv.waitKey(1)
    if c == 27:  # ESC
        break
cv.waitKey(0)
cv.destroyAllWindows()

File: pi_code/pi_camera/camera_test.cpp
#include <iostream>
#include <opencv2/opencv.hpp>

using namespace cv;
using namespace std;

int main() {
    string path = "rstp://192.168.1.154:8554/stream1";
    VideoCapture cap(path);

    if (!cap.isOpened()) {
        cout << "Failed to open camera." << endl;
        return -1;
    }

    int i = 0;
    while (true) {
        Mat frame;
        if (!cap.read(frame)) {
            cout << "Failed to read frame." << endl;
            break;
        }
        imshow("frame", frame);
        i++;

        char c = waitKey(1);
        if (c == 27) { // ESC
            break;
        }
    }

    cap.release();
    destroyAllWindows();

    return 0;
}
